libcaf_net/caf/net/tcp_accept_socket.cpp

```diff
@@ -124,12 +124,8 @@ make_tcp_accept_socket(const uri::authority_type& node, bool reuse_addr) {
     return make_error(sec::cannot_open_port, "no local interface available",
                       to_string(node));
   // Prefer ipv6 addresses.
-  std::stable_sort(std::begin(addrs), std::end(addrs),
-                   [](const ip_address& lhs, const ip_address& rhs) {
-                     if (lhs.embeds_v4())
-                       return rhs.embeds_v4() ? lhs < rhs : false;
-                     return rhs.embeds_v4() ? true : lhs < rhs;
-                   });
+  std::stable_partition(std::begin(addrs), std::end(addrs),
+                        [](const ip_address& ip) { return !ip.embeds_v4(); });
```

Comment:
Yeah, that's definitely a better way to do it. Good catch! 🙂

Reply:
There's one more place where we reorder for ipv6. I'll push the changes.